### PR TITLE
Reducer for CustomizeForm

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

> Adding reducer for CustomizeForm component

### CODE REVIEW

**Comment by upalatucci** [comment]
@glekner @yaacov @metalice  @avivtur pr n. 2 !!

**Comment by yaacov** [comment]
hmm, if  you need a reducer somthing is wrong with the logic :-(
it should goes like this, all logic is called from the onsubmit method:

form renders:
a - user fill in the form (you have 3 distinct versions of form)

form is submitted (onsubmit is called):
b - using backend logic (e.g. template proccess + dryrun) creates a template + parameters, and extract vm object
c - update name + namespace in the vm object, validate using backaend, e.g. vm create + dryrun)

when regular storage is required:
d - if needed the datvolumetemplate is updated, validate using backaend, e.g. vm create + dryrun)
e - if needed add drivers cdrom, validate using backaend, e.g. vm create + dryrun)

when cdrom storage is required:
f - if needed remove boot disk and add boot cdrom, validate using backaend, e.g. vm create + dryrun)
g - if needed add empty disk, validate using backaend, e.g. vm create + dryrun)
f - if needed add drivers cdrom, validate using backaend, e.g. vm create + dryrun)

**Comment by upalatucci** [comment]
Hei @yaacov, 
I can confirm that the flow is exactly as you described. 
I'm not using here a Redux reducer but a useReducer hook by react only to organize better the code. (if is clear to me what are you pointing out).

There were a lot of `setState` for different react states, especially in the `onSubmit` function of the `CustomizeForm` component.
Also, I implemented a basic validation like 'fill the required fields' and 'the volume should not be 0' (in the next pr), and managing this with a `useState` hook could be bad. 
For these reasons, I thought a reducer could be a good fit for the situation. 

I hope I clarified the situation, but if there are further doubts about the use of the reducer or the validation made here, I would be happy to edit my code following suggestions.

**Comment by yaacov** [comment]
> There were a lot of setState for different react states, especially in the onSubmit

no state is needed, if you have a state it's bad :-(

Form 1 (no storage):
a - you get a template
b - you fill in the parameters
no state needed

Form 2 (simple storage)
a - you get a template
b - you fill in the parameters
c - you edit the storage
no state needed

Form 3 (CD storage)
a - you get a template
b - you fill in the parameters
c - you remove the storage
d - you add CD storage
e - you add Empty stroage
no state needed

What am I missing, why do you need state in any of the 3 forms ?

**Comment by glekner** [comment]
@yaacov he is talking about the states of the input fields.
reducer can help making a cleaner code, its not a Redux reducer as he said

We can either think about form validation from scratch for the entire plugin, or let each dev create his own way of thinking about forms and validation, but we need to stick to it

File: src/views/catalog/customize/components/FieldGroup.tsx [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
 };
 
-export const FieldGroup: React.FC<FieldGroupProps> = ({ field, showError }) => {
-  const { t } = useKubevirtTranslation();
+export const FieldGroup: React.FC<FieldGroupProps> = React.memo(({ field, error }) => {
   const { name, description, displayName, required, value: initialValue } = field;
   const [value, setValue] = React.useState(initialValue || '');
 
-  const validated = showError && !value ? 'error' : 'default';
+  const validated = error ? 'error' : 'default';
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply src/views/catalog/customize/components/FieldGroup.tsx:16]: Great thanks

**Pull request closed without merging**